rpadmin/admin.go

```diff
@@ -190,10 +197,22 @@ func newAdminAPI(urls []string, auth Auth, tlsConfig *tls.Config, dialer DialCon
 	a.retryClient.Transport = transport
 	a.oneshotClient.Transport = transport
 
+	if err := a.initURLs(urls, tlsConfig, forCloud); err != nil {
+		return nil, err
+	}
+
+	return a, nil
+}
+
+func (a *AdminAPI) initURLs(urls []string, tlsConfig *tls.Config, forCloud bool) error {
+	if len(a.urls) != len(urls) {
+		a.urls = make([]string, len(urls))
+	}
```

Comment:
Should this be instead if len(a.urls) == 0?

I wonder what is the expected result if len(a.urls) = 1, and len(urls) = 3... We will end up overwriting a.urls[0] and not sure if it's needed.

Reply:
`a.urls` could be initialized here, to original passed in urls to `NewClient()` which could be any number of elements.. This new code for `initURLs()` was moved from constructor to a helper function since it's used in two places now... It initializes _our_ `a.urls`... So we want to overwrite, which the loop will do... But in case the internal slice is not the correct size, we need to make it the correct size, so we check for equivalence, because the loop uses direct index `a.urls[i] = `.

Reply:
I see, thanks!